apps/ws/src/GameManager.ts

```diff
@@ -171,6 +172,44 @@ export class GameManager {
 
         SocketManager.getInstance().addUser(user, gameId);
       }
+
+      if (message.type === RESIGN) {
+        SocketManager.getInstance().broadcast(
+          message.payload.gameId,
+          JSON.stringify({
+            type: GAME_OVER,
+            payload: {
+              result:
+                message.payload.myColor === 'black' ? 'RESIGN_B' : 'RESIGN_W',
+            },
+          }),
+        );
+      }
+
+      if (message.type === OFFER_DRAW) {
+        SocketManager.getInstance().broadcast(
+          message.payload.gameId,
+          JSON.stringify({
+            type: OFFER_DRAW,
+            payload: {
+              from: message.payload.myColor,
+              id: message.payload.myId,
+            },
+          }),
+        );
+      }
+
+      if (message.type === DRAW_OFFER_ACCEPTED) {
+        SocketManager.getInstance().broadcast(
```

Comment:
Are we checking if the other user has offered a draw before accepting it?
It should be stored somewhere that  there was an offer made

Reply:
good one. I thought better do this way since doing db was a lot of hassle. Happy to change if needed? is it?

Reply:
@hkirat how about a in memory store the expires every 30s?

Reply:
In memory yeah makes sense